Partition (WBP)/Functions/VI0039C.VTI0039.fun.yml

```diff
@@ -58,14 +66,705 @@
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Restoration failed') ('') )
+    Endif
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Check for table VTLI0038D, if the update has been successful') ('') )
+    If (#Com_Owner.Record_From_TableD_Is_Update_Successful( #LF_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Update failed') ('') )
+    Endif
+    
+    * UPDATE C
+    * COMMIT
+    * UPDATE D
+    * FETCH C - check updated row exists
+    * FETCH D - check updated row exists
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Updating record of table VTLI0039C') ('') )
+    If (#Com_Owner.Update_Record_Of_TableC_Is_Successful( #LF_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Record does not exists') ('') )
+    Endif
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Rolling back the changes') ('') )
+    Commit
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Updating record of table VTLI0039D') ('') )
+    If (#Com_Owner.Update_Record_Of_TableD_Is_Successful( #LF_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Update failed') ('') )
+    Endif
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Check for table VTLI0038C, if the update has been successful') ('') )
+    If (#Com_Owner.Record_From_TableC_Is_Update_Successful( #LF_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Update failed') ('') )
+    Endif
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Check for table VTLI0038D, if the update has been successful') ('') )
+    If (#Com_Owner.Record_From_TableD_Is_Update_Successful( #LF_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Update failed') ('') )
+    Endif
+    
+    Endif
+    Endroutine
+    
+    *
+    * This routine is responsible to validate the tests for DBMS Optimize for tables VTLI0039C and VTLI0039D
+    *
+    Mthroutine Name(mthDbmsOptimize)
+    Define_Map For(*BOTH) Class(#BIFA002A) Name(#this_rc)
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- DBMS Optimize tests for DataBase type ' + #LF_DB_Type) '' )
+    
+    #Com_Owner.mthClearDataFromTables( #this_rc )
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- For table VTLI0039D, inserting 101 records') '' )
+    
+    Use Builtin(set_session_value) With_Args(DBMS_OPTIMIZE BEGIN_SYNC_100)
+    Begin_Loop Using(#LF_i) To(101)
+    #BIFP0700A := #LF_i
+    Insert Fields(*ALL) To_File(VTLI0039D)
+    Use Builtin(set_session_value) With_Args(DBMS_OPTIMIZE SYNC_POINT)
+    End_Loop
+    Rollback
+    Use Builtin(set_session_value) With_Args(DBMS_OPTIMIZE END_SYNC)
+    
+    * Test that row 100 is in table but 101 is not in table.
+    #Com_Owner.mthAdd_SubMsg( OK ('- For table VTLI0039D, check if 101th record does not exists') '' )
```

Comment:
You haven't tested that row 100 is in the table.

Reply:
done